www/js/app.js

```diff
@@ -1046,20 +1043,18 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
                     }
                 }
                 if (selectedStorage === null) {
-                    alert("Unable to find which device storage corresponds to directory " + archiveDirectory);
+                    uiUtil.systemAlert("Unable to find which device storage corresponds to directory " + archiveDirectory, "Error: no matching storage");
                 }
-            }
-            else {
+            } else {
                 // This happens when the archiveDirectory is not prefixed by the name of the storage
                 // (in the Simulator, or with FxOs 1.0, or probably on devices that only have one device storage)
                 // In this case, we use the first storage of the list (there should be only one)
                 if (storages.length === 1) {
                     selectedStorage = storages[0];
-                }
-                else {
-                    alert("Something weird happened with the DeviceStorage API : found a directory without prefix : "
-                        + archiveDirectory + ", but there were " + storages.length
-                        + " storages found with getDeviceStorages instead of 1");
+                } else {
+                    uiUtil.systemAlert("Error: unprefixed directory", "Something weird happened with the DeviceStorage API : found a directory without prefix : "
```

Comment:
You might not have got here yet, but I guess the title and the message need to be swapped here too.

Reply:
Yes, please don't review as of now, I'm not done yet. I'll request a review from both of you within an hour.